content/en/blog/2024/cve-2024-36129/index.md

```diff
@@ -0,0 +1,127 @@
+linkTitle: Security vulnerability in confighttp
+date: 2024-06-05
+author:
+  '[Juraci Paixão Kröhling](https://github.com/jpkrohling) (OpenTelemetry,
+  Grafana Labs), [Pablo Baeyens](https://github.com/mx-psi) (OpenTelemetry,
+  Datadog)'
+cSpell:ignore: confighttp Baeyens OSTIF zstd configgrpc Miroslav Stampar
+---
+
+On our path toward graduation, the OpenTelemetry project is currently undergoing
+a security audit sponsored by the [CNCF](https://www.cncf.io/), facilitated by
+[OSTIF](https://ostif.org/), and performed by
+[7ASecurity](https://7asecurity.com/). During this process, we have received a
+few ideas about things that we could do better, like using specific compiler
+flags when preparing our [OpenTelemetry Collector
+binaries](https://github.com/open-telemetry/opentelemetry-collector-releases/releases).
+On 31 May 2024, we received [a more serious
+report](https://github.com/open-telemetry/opentelemetry-collector/security/advisories/GHSA-c74f-6mfw-mm4v):
+a malicious user could cause a denial of service (DoS) when using a specially
+crafted HTTP or gRPC request. The advisory was assigned the following CVE
+identifier: [CVE-2024-36129](https://nvd.nist.gov/vuln/detail/CVE-2024-36129).
+
+When sending an HTTP request with a compressed payload, the Collector would
+verify only whether the compressed payload is beyond a certain limit, but not
+its uncompressed version. A malicious payload could then send a “[compressed
+bomb](https://en.wikipedia.org/wiki/Zip_bomb)”, causing the Collector to crash.
+
+Similarly, when sending a gRPC request using zstd compression, the decompression
+mechanism would not respect the limits imposed by gRPC, also causing the
+Collector to crash while decompressing the malicious payload.
+
+A few business hours after the report, on 03 June 2024, Collector developers
+were able to reproduce the report related to HTTP and worked together on a fix
+that [got merged the same
+day](https://github.com/open-telemetry/opentelemetry-collector/pull/10289).
+Given the high score of this issue, we’ve decided to hold the release that would
+have happened on the same day, [completing
+it](https://github.com/open-telemetry/opentelemetry-collector-releases/releases/tag/v0.102.0)
+on 04 June 2024 instead.
+
+After the release, we got a confirmation that gRPC with zstd was also affected.
+Within a few business hours of the confirmation, we worked on a fix that also
+[got merged the same
+day](https://github.com/open-telemetry/opentelemetry-collector/pull/10323). [We
+released
+v0.102.1](https://github.com/open-telemetry/opentelemetry-collector-releases/releases/tag/v0.102.1)
+right after that.
+
+## Am I affected?
+
+You are affected by this vulnerability if you have an OpenTelemetry Collector
+with one or more HTTP or gRPC receivers on a public port, such as the OTLP
+Receiver with the “HTTP” or “gRPC” protocol enabled (typically on ports 4318 and
+4317, respectively) AND the receiver has version 0.101.0 or below. The
```

Comment:
No longer true, we have `0.102.1`. Should we just say:
```suggestion
4317, respectively) AND the receiver has version 0.102.0 or below. The
```

Reply:
Technically, for confighttp, it's 0.101.0. But I agree that it might be easier to state a single version everywhere.

Reply:
Yeah, agreed, better to stick to a single version for simplicity